pymc/distributions/distribution.py

```diff
@@ -396,6 +397,16 @@ def __new__(
         if not isinstance(name, string_types):
             raise TypeError(f"Name needs to be a string but got: {name}")
 
+        if transform is None:
+            transform = UNSET
+            default_transform = None
+            warnings.warn(
+                "To disable default transform, please use default_transform=None"
+                " instead of transform=None. Setting transform to None will"
+                " not take effect in future.",
+                UserWarning,
+            )
```

Comment:
We can be a tiny bit more precise here

```suggestion
        if transform is None and default_transform is UNSET:
            default_transform = None
            warnings.warn(
                "To disable default transform, please use default_transform=None"
                " instead of transform=None. Setting transform to None will"
                " not have any effect in future.",
                UserWarning,
            )
```

If the user specified `default_transform`, then we can assume they already know the new API and the warning/magic behavior is not needed. Also I think there's nothing to be gained by setting `transform=UNSET`?